src/rapids_dependency_file_generator/rapids_dependency_file_generator.py

```diff
@@ -199,14 +237,21 @@ def get_filename(file_type, file_prefix, matrix_combo):
     """
     file_type_prefix = ""
     file_ext = ""
+    file_name_prefix = file_key
     if file_type == str(OutputTypes.CONDA):
         file_ext = ".yaml"
-    if file_type == str(OutputTypes.REQUIREMENTS):
+    elif file_type == str(OutputTypes.REQUIREMENTS):
         file_ext = ".txt"
         file_type_prefix = "requirements"
+    elif file_type == str(OutputTypes.PYPROJECT):
+        file_ext = ".toml"
+        # Unlike for files like requirements.txt or conda environment YAML files, which
+        # may be named with additional prefixes (e.g. all_cuda_*) pyproject.toml files
+        # need to have that exact name and are never prefixed.
+        file_name_prefix = "pyproject"
     suffix = "_".join([f"{k}-{v}" for k, v in matrix_combo.items()])
```

Comment:
Could the suffix still accidentally apply to pyproject files here?

Reply:
Hmm that is a good question. Zooming out, I think what this question is really asking is whether we should prevent the matrix key from being provided in one of the files entries if the output type is pyproject. We probably should be preventing that at the schema level, would you agree?

Reply:
Yes, ideally we would prevent it at the schema level, but to not overcomplicate things, I would suggest to just check for it in the code and move forward.

Reply:
Yup I agree with that. I don't want to overburden the schema.

I've added a check in the code and a corresponding test.